packages/console/src/contexts/SubscriptionDataProvider/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { pick } from '@silverhand/essentials';
+
+import { type NewSubscriptionCountBasedUsage } from '@/cloud/types/router';
+import { isCloud } from '@/consts/env';
+
+import { type SubscriptionUsageOptions } from './types';
+
+/* === For new pricing model === */
+const isSubscriptionUsageWithInLimit = <T extends keyof NewSubscriptionCountBasedUsage>(
+  { quotaKey, subscriptionUsage, subscriptionQuota, usage }: SubscriptionUsageOptions<T>,
+  inclusive = true
+) => {
+  // No limitations for OSS version
+  if (!isCloud) {
+    return true;
+  }
+
+  const usageValue = usage ?? subscriptionUsage[quotaKey];
```

Comment:
Better to leave comments about this `usage` param.

Reply:
Comments added.